nova/cmd/manage.py

```diff
@@ -2250,11 +2253,14 @@ def _get_rp_uuid_for_host(ctxt, host):
                 cctxt, host)
 
             if len(nodes) > 1:
-                # This shouldn't happen, so we need to bail since we
-                # won't know which node to use.
-                raise exception.TooManyComputesForHost(
-                    num_computes=len(nodes), host=host)
-            return nodes[0].uuid
+                if 'ironic' not in host:
+                    # Multiple nodes for non-ironic hosts shouldn't happen, so
+                    # we need to bail since we won't know which node to use.
+                    raise exception.TooManyComputesForHost(
+                        num_computes=len(nodes), host=host)
+                else:
+                    return [node.uuid for node in nodes]
```

Comment:
I think it would look cleaner to have this as the last and single return statement.

Reply:
You're right. Changed accordingly.

Reply:
Indentation is one too deep now.